UnitTests/PaymentsCoreTests/APIClientTests.swift

```diff
@@ -0,0 +1,50 @@
+import XCTest
+@testable import PaymentsCore
+
+class APIClientTests: XCTestCase {
+    
+    lazy var apiClient: APIClient = {
+        APIClient(
+            urlSession: URLSession(urlProtocol: URLProtocolMock.self),
+            environment: .sandbox
+        )
+    }()
+
+    func testAccessTokenMockRequest_Success() {
+        let expect = expectation(description: "Get mock response for access token request")
+
+        URLProtocolMock.requestResponses.append(MockAccessTokenRequestResponse(success: true))
+        
+        apiClient.fetch(endpoint: AccessTokenRequest(clientID: "")) { result, correlationId in
+            switch result {
+            case .success(let response):
+                XCTAssertEqual(response?.accessToken, "TestToken")
+            case .failure(let error):
+                XCTFail("Wrong mock response with error: \(error)")
+            }
+
+            expect.fulfill()
+        }
+
+        waitForExpectations(timeout: 1)
+    }
+
+    func testAccessTokenMockRequest_Failure() {
+        let expect = expectation(description: "Get mock response for access token request")
+
+        URLProtocolMock.requestResponses.append(MockAccessTokenRequestResponse(success: false))
+
+        apiClient.fetch(endpoint: AccessTokenRequest(clientID: "")) { result, correlationId in
+            switch result {
+            case .success:
+                XCTFail("Should not be able to successfully decode a result")
+            case .failure(let error):
+              XCTAssertNotNil(error)
```

Comment:
Can we assert on error type here instead of just "not nil"?

Reply:
The `fetch` method should have tests for each error type possible on `CoreError`

Reply:
updated! I think we can create a follow up work item to define how we want to handle networking errors in the SDK. I feel like an `Error` struct makes sense, since we'll have additional context / data that we want to pass back to the merchant, which doesn't lend itself well to using an `enum`

Reply:
I agree - we'll likely want to provide additional context/info and should set up for that 👍